crates/worker/src/partition/snapshots/snapshot_task.rs

```diff
@@ -59,43 +62,32 @@ impl SnapshotPartitionTask {
             )
             .await?;
 
-        let metadata = self.write_snapshot_metadata_header(snapshot).await?;
+        let metadata = self.metadata(&snapshot, SystemTime::now());
 
-        // todo(pavel): SnapshotRepository integration will go in here in a future PR
+        self.snapshot_repository
+            .put(&metadata, snapshot.base_dir)
+            .await
+            .map_err(|e| SnapshotError::RepositoryIo(self.partition_id, e))?;
```

Comment:
Ideally we should, and I was planning to do a separate hardening pass adding retries, but I may as well add it now. We must have a fallback pruning strategy in place anyway, so I was relying on that to catch any partial uploads. But it would definitely be nice not to leave half-uploaded files around. That would become much more important when we are dealing with incremental snapshots where the individual SST files are not clearly tied to any one snapshot.